akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
-        }
-      })
-    }
+      val behavior: Behavior[Command] = (
+        Behaviors.immutablePartial[Command] {
+          case (_, Stop) ⇒ Behaviors.stopped
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
 
-    "signal PostStop after voluntary termination" in {
-      sync(setup("ctx02", ignorePostStop = false) { (ctx, startWith) ⇒
-        startWith.keep { subj ⇒
-          stop(subj)
-        }.expectMessage(expectTimeout) {
-          case (msg, _) ⇒
-            msg should ===(GotSignal(PostStop))
-        }
-      })
+      val actor = spawn(behavior)
+      actor ! Stop
+      probe.expectMessage(GotSignal(PostStop))
     }
 
     "restart and stop a child actor" in {
-      sync(setup("ctx03") { (ctx, startWith) ⇒
-        val self = ctx.self
-        val ex = new Exception("KABOOM2")
-        startWith.mkChild(None, ctx.spawnMessageAdapter(ChildEvent), self) {
-          case (subj, child) ⇒
-            val log = muteExpectedException[Exception]("KABOOM2", occurrences = 1)
-            child ! Throw(ex)
-            (subj, child, log)
-        }.expectMessage(expectTimeout) {
-          case (msg, (subj, child, log)) ⇒
-            msg should ===(ChildEvent(GotSignal(PreRestart)))
-            log.assertDone(expectTimeout)
-            child ! BecomeInert(self) // necessary to avoid PostStop/Terminated interference
-            (subj, child)
-        }.expectMessageKeep(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(BecameInert)
-            stop(subj)
-            ctx.watch(child)
-            ctx.watch(subj)
-        }.expectTermination(expectTimeout) {
-          case (t, (subj, child)) ⇒
-            if (t.ref === child) subj
-            else if (t.ref === subj) child
-            else fail(s"expected termination of either $subj or $child but got $t")
-        }.expectTermination(expectTimeout) { (t, subj) ⇒
-          t.ref should ===(subj)
-        }
-      })
+      val probe = TestProbe[Event]()
+
+      val child: Behavior[Command] = (Behaviors.immutablePartial[Command] {
+        case (_, Fail) ⇒ throw new RuntimeException("Boom")
+        case (_, Inert) ⇒
+          probe.ref ! InertEvent
+          Behaviors.immutablePartial[Command] {
+            case (_, Ping) ⇒
+              probe.ref ! Pong
+              Behaviors.same
+            case _ ⇒ Behaviors.unhandled
+          }
+      } onSignal {
+        case (_, signal) ⇒
+          probe.ref ! GotChildSignal(signal)
+          Behavior.stopped
+      }).decorate
+
+      val parent: Behavior[Command] = (Behaviors.immutablePartial[Command] {
+        case (ctx, MakeChild) ⇒
+          val childRef = ctx.spawnAnonymous(
+            Behaviors.supervise(child).onFailure(SupervisorStrategy.restart)
+          )
+          ctx.watch(childRef)
+          probe.ref ! ChildMade(childRef)
+          Behavior.same
```

Comment:
I think we can skip the `MakeChild` message, and just create the child on actor start with `Behaviors.setup` in the tests that just spawn one child and interact with it, to cut some boilerplate.

As we anyways close over the `TestProbe` the setup logic can send the ref to the test without any other message interaction.